openlibrary/templates/books/custom_carousel_card.html

```diff
@@ -32,12 +32,14 @@
 $ work_key = book['key'] if book['key'].startswith('/work') else book.get('work_key')
 
 <div class="book carousel__item">
-  <div class="book-cover">
+  <div class="book-cover loading-gradient">
     <a href="$(url)" data-ol-link-track="BookCarousel|CoverClick|$key">
       $if cover_url:
         <img class="bookcover" loading="lazy"
           title="$('%s%s'%(title,byline))"
           alt="$('%s%s'%(title,byline))"
+          style="opacity: 0; transition: opacity 0.5s ease-in-out;"
+          onload="this.closest('.loading-gradient')?.classList.remove('loading-gradient'); this.style.opacity = 1;"
```

Comment:
This looks soooooo satisfying :P As you noted, we should toss this into the JS file. We'll kind of treat `loading-gradient` as a "CSS/JS Component" -- i.e. a component that is activated via a CSS class, and has some JS logic and CSS associated with it. Then we can have all the loading-gradients behave in this lovely manner!

Ray notes that these don't have a width/height, causing the loading-gradient not to appear in the slick carousels. My guess would be this is because of slick lazy loading things in. I think what you've got here makes sense, we might have to have 2 JS listeners. This basically means the loading-gradient component can be added to either a direct img tag, or a parent of an image, and the JS/CSS should respond and handle both cases.